package/MDAnalysis/coordinates/PDB.py

```diff
@@ -1080,13 +1082,18 @@ def get_attr(attrname, default):
         altlocs = get_attr('altLocs', ' ')
         resnames = get_attr('resnames', 'UNK')
         icodes = get_attr('icodes', ' ')
+        chainids = get_attr('chainIDs', ' ')
         segids = get_attr('segids', ' ')
         resids = get_attr('resids', 1)
         occupancies = get_attr('occupancies', 1.0)
         tempfactors = get_attr('tempfactors', 0.0)
         atomnames = get_attr('names', 'X')
         elements = get_attr('elements', ' ')
         record_types = get_attr('record_types', 'ATOM')
+        chainids = (
+            [id[:4] if (chainids[index] != '') else segids[index][-1:]
+            for index, id in enumerate(chainids)]
+            ) # check for a chainID, if no chainID default to segid
```

Comment:
I find this not ideal -- there may be instances where the chain is empty in a file that otherwise has L/M/N chains. In these cases, we wouldn't want to muddy the waters by assigning only *some* atoms to the last letter of the `segid`. Instead, it would be more beneficial to simply check if the atoms have the chainID attribute, and if not, use the segid for all of them.

> id[:4]

Could you explain why you decided to limit the chainID to 4 letters? In the PDB spec, chainIDs are only 1 character long.

Reply:
this makes sense. Considering right now

Reply:
Yeah, I think I overthought this one. hasattr would be a much better use case. Also, I don't have a great reason for limiting to 4 characters, I think I misread. Working on the change :+1: